### ISSUE TITLE
StableDiffusionImg2ImgPipeline creates different outputs when using a list of generators

### ISSUE THREAD

**Issue by d1g1t** [huggingface/diffusers#2560]
### Describe the bug

When passing a list of generators, `StableDiffusionImg2ImgPipeline` creates different images based on the position of the generator in the list.

The reproduction code generates 4 completely different images even though only two seeds were passed, but in the different order. It is however consistent in that it will generate the same 4 images if run again.

Passing the same seed twice in the list also results in two different images

### Reproduction

```python
#!/usr/bin/python
import torch
import requests
from PIL import Image
from io import BytesIO

from diffusers import StableDiffusionImg2ImgPipeline

pipe = StableDiffusionImg2ImgPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16, local_files_only=True, safety_checker=None)
pipe = pipe.to("cuda")

generator1 = torch.Generator(device='cuda').manual_seed(1001)
generator2 = torch.Generator(device='cuda').manual_seed(1002)

prompt = "Labrador in the style of Vermeer"

url = "https://raw.githubusercontent.com/CompVis/stable-diffusion/main/assets/stable-samples/img2img/sketch-mountains-input.jpg"

response = requests.get(url)
init_image = Image.open(BytesIO(response.content)).convert("RGB")
init_image.thumbnail((512, 512))

images = pipe([ prompt, prompt ], image=[ init_image, init_image ], strength=0.75, guidance_scale=7.5, generator=[ generator1, generator2 ], num_inference_steps=20).images

for idx,image in enumerate(images):
	image.save(f"test_{idx}.png")

images = pipe([ prompt, prompt ], image=[ init_image, init_image ], strength=0.75, guidance_scale=7.5, generator=[ generator2, generator1 ], num_inference_steps=20).images

for idx,image in enumerate(images):
	image.save(f"test_{idx+2}.png")
```

### System Info

- `diffusers` version: 0.14.0
- Platform: Linux-5.15.0-67-generic-x86_64-with-glibc2.35
- Python version: 3.10.6
- PyTorch version (GPU?): 1.13.1+cu117 (True)
- Huggingface_hub version: 0.12.1
- Transformers version: 4.26.1
- Accelerate version: 0.17.0.dev0
- xFormers version: 0.0.16
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

**Comment by patrickvonplaten** [comment]
Hey @d1g1t,

Thanks for the issue! This is a common misunderstanding how PyTorch's generators work and also arguably a bad design choice from PyTorch.

In short, PyTorch generators are advanced whenever they are used by a function such as `torch.randn(...)` this means if you do:

```py
import torch

generator = torch.manual_seed(0)

x_1 = torch.randn((10,10), generator=generator)

x_2 = torch.randn((10,10), generator=generator)
```

you get

```py
assert (x_1 - x_2).abs().max() > 0.1
```

So PyTorch's generators are stateful (in comparison Jax random seed handling is stateless which is better IMO: https://jax.readthedocs.io/en/latest/jep/263-prng.html) 

So to fix your code example you should do:

```py
#!/usr/bin/python
import torch
import requests
from PIL import Image
from io import BytesIO

from diffusers import StableDiffusionImg2ImgPipeline

pipe = StableDiffusionImg2ImgPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", torch_dtype=torch.float16, local_files_only=True, safety_checker=None)
pipe = pipe.to("cuda")

generator1 = torch.Generator(device='cuda').manual_seed(1001)
generator2 = torch.Generator(device='cuda').manual_seed(1002)

prompt = "Labrador in the style of Vermeer"

url = "https://raw.githubusercontent.com/CompVis/stable-diffusion/main/assets/stable-samples/img2img/sketch-mountains-input.jpg"

response = requests.get(url)
init_image = Image.open(BytesIO(response.content)).convert("RGB")
init_image.thumbnail((512, 512))

images = pipe([ prompt, prompt ], image=[ init_image, init_image ], strength=0.75, guidance_scale=7.5, generator=[ generator1, generator2 ], num_inference_steps=20).images

for idx,image in enumerate(images):
	image.save(f"test_{idx}.png")

### IMPORTANT: create generators again here
generator1 = torch.Generator(device='cuda').manual_seed(1001)
generator2 = torch.Generator(device='cuda').manual_seed(1002)

images = pipe([ prompt, prompt ], image=[ init_image, init_image ], strength=0.75, guidance_scale=7.5, generator=[ generator2, generator1 ], num_inference_steps=20).images

for idx,image in enumerate(images):
	image.save(f"test_{idx+2}.png")
```

**Comment by d1g1t** [comment]
Ah, That makes sense. Thanks for clarifying.
